library/math/numbersLine/src/components/Jump.tsx

```diff
@@ -6,40 +6,59 @@ import MoveableElement from "./MoveableElement";
 import { useNumbersLineContext } from "@/context/numbersLineContext";
 import { calculatRulerWidth, calculatUnitsAmount } from "../lib/utils";
 import { RulerPadding, jumpArrowHeight } from "../consts/elementConsts";
-import { MatchBaseJumpClassName, claculateHeightStartingPositionElement, claculateWidthStartingPositionElement } from "../lib/stylesUtils";
+import { MatchBaseJumpClassName, clacHeightStartPosition, clacWidthStartPosition } from "../lib/stylesUtils";
+import { deleteDragElement, duplicateDragElement } from "../hooks/useAction";
 
 interface IProps {
   element: IElement;
 }
 
 const Jump = ({ element }: IProps) => {
-  const { windowSize, type, dragElements, setDragElements, idDraggElementClick } = useNumbersLineContext();
+  const { windowSize, type, dragElements, setDragElements, idDraggElementClick, setIdDraggElementClick } = useNumbersLineContext();
   const [unit, setUnit] = useState(windowSize.width / calculatUnitsAmount(type));
   const [isJumpUnderRuler, setIsJumpUnderRuler] = useState(false);
-  const targetRef = React.useRef<any>(null);
+  const moveableRef = React.useRef<any>(null);
 
   useEffect(() => {
     let rulerWidth = calculatRulerWidth(windowSize.width, RulerPadding) / calculatUnitsAmount(type);
     setUnit(rulerWidth);
   }, [type, windowSize.width]);
 
+  useEffect(() => {
+    const root = document.documentElement;
+    root.style.setProperty("--margin-value", `${isJumpUnderRuler ? -50 : 28}px`);
+  }, [isJumpUnderRuler]);
+
   const changeHidenumbers = () => {
     let newElements = dragElements.map((item: IElement) => (item.id === idDraggElementClick ? { ...item, hideNumber: !item.hideNumber } : item));
     setDragElements(newElements);
   };
 
+  const ableProps = {
+    ButtonViewable: true,
+    deleteButtonViewable: true,
+    onDeleteClick: () => setDragElements(deleteDragElement(element.id, dragElements)),
+    copyButtonViewable: true,
+    onCopyClick: () => {
+      setDragElements(duplicateDragElement(element.id, dragElements));
+      setIdDraggElementClick("");
+    },
+    widthElement: unit * element.value,
+    underRuler: isJumpUnderRuler,
+  };
```

Comment:
remove unnessecry values and do this const genery

Reply:
use type IAbleprops you maked. and you can declare this type on moveableItem component because delete and copy are common to all elements.